File: src/main/java/app/application/utils/YoutubeDownloadService.java
package app.application.utils;


import app.application.listener.YoutubeDownloadListener;
import com.github.kiulian.downloader.YoutubeDownloader;
import lombok.Setter;
import org.springframework.beans.factory.annotation.Autowired;

public abstract class YoutubeDownloadService {

    @Autowired
    protected YoutubeDownloader youtubeDownloader;

    @Autowired
    protected UserConfigHandler userConfigHandler;

    @Autowired
    protected DialogManager dialogManager;

    @Setter
    protected YoutubeDownloadListener youtubeDownloadListener;

}

File: src/main/kotlin/app/application/spring/service/download/YoutubeDownloadService.kt
package app.application.spring.service.download

import app.application.listener.YoutubeDownloadListener
import app.application.spring.service.DialogManager
import app.application.spring.service.UserConfigHandler
import com.github.kiulian.downloader.YoutubeDownloader
import org.springframework.beans.factory.annotation.Autowired

abstract class YoutubeDownloadService{
    @Autowired
    protected lateinit var youtubeDownloader: YoutubeDownloader

    @Autowired
    protected lateinit var userConfigHandler: UserConfigHandler

    @Autowired
    protected lateinit var dialogManager: DialogManager

    var youtubeDownloadListener: YoutubeDownloadListener? = null

}